### PR TITLE
Added fixes for the filename exported and layer name in QGIS

### PR DESCRIPTION
**Pull request by mtravis**
Added some logic provide fixes for #8 and #28 

Had a little help from Cursor AI with this one.

### CODE REVIEW

**Comment by cholmes** [comment]
Awesome! I'll try this out tomorrow, thanks for coding it up!

File: qgis_plugin_gpq_downloader.py [hunk 1]
```diff
@@ -661,36 +662,54 @@ def run(self, default_source=None):
...
+    def handle_validation_complete(self, success, message, validation_results, url, extent, dialog):
+        """Handle validation completion and start download if successful."""
         if success:
-            # Generate default filename
-            from datetime import datetime
-            timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
-            default_filename = f"geoparquet_download_{timestamp}.parquet"
-            default_save_path = str(self.download_dir / default_filename)
+            # Get current date for filename
+            current_date = datetime.datetime.now().strftime('%Y%m%d')
```
Comment (cholmes) [inline qgis_plugin_gpq_downloader.py:676]: Curious why you shortened this to just be year, month, day. Just to make it a bit shorter? 

I think I could be convinced of it, but the benefit of hour, minute, second is that if I download more than one file of one type I can just not think / edit. But if I pull down overture_buildings twice then it'll ask me to enter another name if it's still the same day.

File: qgis_plugin_gpq_downloader.py [hunk 2]
```diff
@@ -661,36 +662,54 @@ def run(self, default_source=None):
...
+                if theme == "base":
+                    subtype = dialog.base_subtype_combo.currentText()
+                    filename = f"overture_base_{subtype}_{current_date}.parquet"
+                else:
+                    filename = f"overture_{theme}_{current_date}.parquet"
+            
+            elif dialog.sourcecoop_radio.isChecked():
+                selection = dialog.sourcecoop_combo.currentText()
+                # Convert display name to safe filename format
+                safe_name = selection.lower().replace(" ", "_").replace("/", "_")
```
Comment (cholmes) [inline qgis_plugin_gpq_downloader.py:690]: One little observation here is that the vida name became quite long. But we could also just use this to nudge us to shorter names...

File: qgis_plugin_gpq_downloader.py [hunk 3]
```diff
@@ -661,36 +662,54 @@ def run(self, default_source=None):
...
+                filename = f"sourcecoop_{safe_name}_{current_date}.parquet"
+            
+            else:  # custom URL
+                filename = f"custom_download_{current_date}.parquet"
```
Comment (cholmes) [inline qgis_plugin_gpq_downloader.py:694]: I think this works for the first iteration, but I do think we should add a ticket to try to get the file name of the remote url used. At least if it's a .parquet file, which is most of what I do with 'custom' (as opposed to the glob (*) urls - it'd be hard to automatically get a file name from those).

**Review by cholmes — APPROVED** [review]
Looks good! I put up a couple comments, and I think we should make an issue out of one (to make the custom download a bit better). But this is great! I'll approve, feel free to merge if you have permissions (though not sure if you do, for the wiki I think I just marked you as 'collaborator' as that was the only option, and not sure if that's merge/write for the repo - but I'll add you after this). Merging as is is fine, or feel free to improve a bit based on suggestions and merge.

File: qgis_plugin_gpq_downloader.py (hunk 3 above)
Comment (mtravis) [reply qgis_plugin_gpq_downloader.py:694]: Ah, I hadn't really taken the custom urls into consideration, was looking more at what comes out via Overture/SC

Lets pick that up in new ticket as suggested.

File: qgis_plugin_gpq_downloader.py (hunk 1 above)
Comment (mtravis) [reply qgis_plugin_gpq_downloader.py]: This was done in error. Makes sense to keep it as a full timestamp. I've reverted to the original.

**Pull request merged**